### DISCUSSION TITLE
Autokeras Best Practice for structured Data

### DISCUSSION THREAD

**Discussion by NKlinke** [keras-team/autokeras#1398] [Q&A] [answered]
Hi there,

first of all: Thanks for this gorgeous python package. I am looking forward to your future work!

I have been using Autokeras to fit regression and classification models to structured data. Working with the module, some questions came up about best practices, i would like to get some feedback on. 

1. Data Preprocessing
a. Is Autokeras automatically doing some data preprocessing or do I have to take care of normalization manually? So far i used the data as is. But on the current use case numbers are quite different, so i was wondering. 
b. Is Autokeras taking care of encoding of categorical variables automatically? If so, how are they being encoded?

2. Training combined or separate models
I have several outputs that i want to train. Does it make sense to train one net for multiple outputs, even if the physical meaning of the outputs might be very different? Currently I rather train independently because test errors were lower. 

3. Monitoring
I would like to monitor training process. Printing with verbose=1 is a bit too much, but using verbose=0 does not give me any output. Best would be to get the "Training Summary": 
![image](https://user-images.githubusercontent.com/17301532/97010564-daa36480-1545-11eb-8c3f-32c6b721a9a6.png)

Kind Regards
NKlinke

**Comment by haifeng-jin** [comment, marked as answer]
1.Currently, we support data normalization and categorical encoding (encoding strings into 0, 1, 2, 3...).
2. Yes. You should use the multi-modal and multi-task tutorial for that.
3. We are highly rely on Keras Tuner for controlling the output, which is working on that.

**Reply by NKlinke** [reply]
Thank you very much.
